sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/policy/TextReplacementPolicy.java

```diff
@@ -86,7 +86,9 @@ public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineN
         return next.process()
             .doOnError(throwable -> {
                 networkCallRecord.setException(new NetworkCallError(throwable));
-                recordedData.addNetworkCall(networkCallRecord);
+                if (recordedData != null) { // when using testproxy, recordedData will be set to null
```

Comment:
If recordedData is null, what TextReplacementPolicy.java does in test proxy setup?

Reply:
I had the same thoughts that I had shared with @haolingdong-msft too. We should not be depending on recordedData/networkCall as with TestProxyTestBase we never initialize this data.